### PR TITLE
Security fixes for 8.2.0

### PR DESCRIPTION
**Pull request by hugovk**
See commits and release notes for details.

(Plus reorder release notes to roughly alphabetical, nothing requiring promoting to the top.)

### CODE REVIEW

**Comment by hugovk** [comment]
@wiredfool There are 3 fails (in addition to xfails) on the valgrind workflow, is this good to merge or do we want to check those first?

File: src/libImaging/Jpeg2KDecode.c [hunk 1]
```diff
@@ -814,23 +814,40 @@ j2k_decode_entry(Imaging im, ImagingCodecState state) {
...
         tile_height = tile_info.y1 - tile_info.y0;
-        components = tile_info.nb_comps == 3 ? 4 : tile_info.nb_comps;
-        if ((tile_width > UINT_MAX / components) ||
-            (tile_height > UINT_MAX / components) ||
-            (tile_width > UINT_MAX / (tile_height * components)) ||
-            (tile_height > UINT_MAX / (tile_width * components))) {
+
+        /* Total component width = sum (component_width) e.g, it's
+         legal for an la file to have a 1 byte width for l, and 4 for
+         a. and then a malicious file could have a smaller tile_bytes
```
Comment (radarhere) [inline src/libImaging/Jpeg2KDecode.c:832]:
```suggestion
         a, and then a malicious file could have a smaller tile_bytes
```

**Comment by wiredfool** [comment]
I know these patches are good, and I'm not too worried because they're 3 separate images in the fuzz_images test, which is mainly ensuring that the fuzzer runs under test (but not as fuzzing).

**Comment by hugovk** [comment]
Thanks, proceeding with the release!

**Comment by guitos** [comment]
Hi, the CVEs mentioned in commits and release notes will be reported to Mitre?
Thanks.

**Comment by radarhere** [comment]
They have been, yes. Go to https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#security to see links to the reports.

**Comment by tcullum-rh** [comment]
Why are there 2 CVEs assigned for "Fix OOB read in Jpeg2KDecode?" Could someone please explain?

**Comment by hugovk** [comment]
One each for:

* CVE-2021-25287: "There is an out-of-bounds read in J2kDecode, in j2ku_graya_la."

* CVE-2021-25288: "There is an out-of-bounds read in J2kDecode, in j2ku_gray_i."

---

The CVEs from https://pillow.readthedocs.io/en/stable/releasenotes/8.2.0.html#security at MITRE ([example](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-25288)) are still showing as reserved:

> ** RESERVED ** This candidate has been reserved by an organization or individual that will use it when announcing a new security problem. When the candidate has been publicized, the details for this candidate will be provided.

Does we or Tidelift need to do anything to publicise them?

**Comment by radarhere** [comment]
https://cve.mitre.org/cve/researcher_reservation_guidelines#researcher_reservation_guidelines#12
> 12. Notify the CVE Team that the vulnerability has been made public using the CVE Request web form, and selecting "Notify CVE about a Publication."

I've submitted https://cveform.mitre.org/ accordingly.

**Comment by wiredfool** [comment]
There were 2 cves assigned for 2 fuzzier test cases that looked very similar before I got to fixing it.  Once the fix was done, there was one bug that could hit any multiband mode but the fuzzier only found these two.

**Comment by radarhere** [comment]
I received an e-mail response, and the CVEs should now be public.

**Pull request merged**